databricks/koalas/indexing.py

```diff
@@ -1147,6 +1147,13 @@ def _select_cols_by_iterable(
                 (self._internal.spark_frame.select(col).columns[0],) for col in cols_sel
             ]
             data_spark_columns = list(cols_sel)
+        elif all(isinstance(key, bool) for key in cols_sel):
+            column_labels = [
```

Comment:
We need to check the length of the boolean list?

```py
>>> pdf = pd.DataFrame({"a": [1, 2], "b": [3, 4], "c": [5, 6]}, index=["x", "y"])
>>> pdf.loc[:, [True, False, True]]
   a  c
x  1  5
y  2  6
>>> pdf.loc[:, [True, False]]
Traceback (most recent call last):
...
IndexError: Boolean index has wrong length: 2 instead of 3
>>> pdf.loc[:, [True, False, True, True]]
Traceback (most recent call last):
...
IndexError: Boolean index has wrong length: 4 instead of 3
```

Reply:
Good catch! Updated.

Reply:
```
>>> kdf.loc[:, np.array([True, False])]
Traceback (most recent call last):
...
IndexError: Boolean index has wrong length: 2 instead of 3
```